data_model/src/events/data/events.rs

```diff
@@ -5,35 +5,38 @@ use iroha_primitives::small::SmallVec;
 
 use super::*;
 
+/// Generic [`MetadataChanged`] struct.
+/// Depending on wrapping event could mean inserted or removed metadata `(key, value)` pair.
+#[derive(
+    Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Decode, Encode, Deserialize, Serialize, IntoSchema,
+)]
+#[allow(missing_docs)]
+pub struct MetadataChanged<ID> {
+    pub origin_id: ID,
+    pub key: Name,
+    pub value: Box<Value>,
```

Comment:
Box to make enum variants equally sized.

Removing value could be considered since we could query this value, however removed value couldn't be observed this way.

I took approach by representing opposite events by single struct and distinguish what actually happens by wrapping event to reduce code duplication.

Reply:
What `enum`? If this boxing is needed for a structure that houses this object, you should at least mention this object in the comments. Moreover, why isn't this entire comment a literal comment in-code?

Reply:
why do we care that they are equally sized? also we actually have a `ValueBox` type

Reply:
Boxing seems to be the most sensible option. `WorldEvent`s are released quite frequently, meaning that if we don't box, we'd be wasting a lot of extra space.